src/effects/Skid.tsx

```diff
@@ -20,21 +19,26 @@ interface SkidProps {
 
 export function Skid({ count = 500, opacity = 0.5, size = 0.4 }: SkidProps): JSX.Element {
   const ref = useRef<InstancedMesh>(null)
-  const { wheels, chassisBody } = useStore((state) => state)
+  const [chassisBody, wheels] = useStore((state) => [state.chassisBody, state.wheels])
 
-  let controls: Controls
+  let brake: Controls['brake']
   let index = 0
   useFrame(() => {
-    controls = getState().controls
-    if (controls.brake && mutation.speed > 10) {
-      e.setFromRotationMatrix(m.extractRotation(chassisBody.current!.matrix))
-      setItemAt(ref, e, wheels[2].current!.getWorldPosition(v), index++)
-      setItemAt(ref, e, wheels[3].current!.getWorldPosition(v), index++)
+    brake = getState().controls.brake
+    if (chassisBody.current && wheels[2].current && wheels[3].current && brake && mutation.speed > 10) {
+      e.setFromRotationMatrix(m.extractRotation(chassisBody.current.matrix))
+      setItemAt(ref.current!, e, wheels[2].current, index++)
+      setItemAt(ref.current!, e, wheels[3].current, index++)
       if (index === count) index = 0
     }
   })
 
-  useLayoutEffect(() => void ref.current!.geometry.rotateX(-Math.PI / 2), [])
+  useLayoutEffect(() => {
+    ref.current!.geometry.rotateX(-Math.PI / 2)
+    return () => {
+      ref.current!.geometry.rotateX(Math.PI / 2)
+    }
```

Comment:
why is this needed?

Reply:
Every time the component renders the layoutEffect call rotates the geometry, so every render it rotates an additional 90 degrees. This rotates it back when the component unmounts so that it's always the correct rotation.

It would be nice to be able to set the rotation in the declaration of the geometry or being able to set the rotation directly, but I couldn't see a way to do that.

Reply:
Ahhh thanks for the explanation!

Can't you pass rotate as a prop to instanced mesh? or to planeGeometry?
Or does that not work for instanced meshes?

Reply:
I couldn't see a way to do that.

Reply:
like this:

```jsx
<planeGeometry rotation={[x,y,z]} />
```

https://docs.pmnd.rs/react-three-fiber/API/objects